features/admin.application-templates.v1/pages/application-template.tsx

```diff
@@ -0,0 +1,108 @@
+import ApplicationTemplateMetadataProvider from
+    "../provider/application-template-metadata-provider";
+import ApplicationTemplateProvider from "../provider/application-template-provider";
+
+/**
+ * Props for the Applications templates page.
+ */
+type ApplicationTemplateSelectPageInterface = TestableComponentInterface;
+
+/**
+ * Choose the application template from this page.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns Application template select page.
+ */
+const ApplicationTemplateSelectPage: FunctionComponent<ApplicationTemplateSelectPageInterface> = ({
+    "data-testid": testId = "application-templates"
+}: ApplicationTemplateSelectPageInterface): ReactElement => {
+
+    const { t } = useTranslation();
+
+    const [ showWizard, setShowWizard ] = useState<boolean>(false);
+    const [ selectedTemplate, setSelectedTemplate ] = useState<ExtensionTemplateListInterface>(null);
+
+    /**
+     * Handles back button click.
+     */
+    const handleBackButtonClick = (): void => {
+        history.push(AppConstants.getPaths().get("APPLICATIONS"));
+    };
+
+    return (
+        <PageLayout
+            pageTitle="Register New Application"
+            title={ t("console:develop.pages.applicationTemplate.title") }
+            contentTopMargin={ true }
+            description={ t("console:develop.pages.applicationTemplate.subTitle") }
+            backButton={ {
+                "data-testid": `${ testId }-page-back-button`,
+                onClick: handleBackButtonClick,
+                text: t("console:develop.pages.applicationTemplate.backButton")
+            } }
+            titleTextAlign="left"
+            bottomMargin={ false }
+            showBottomDivider
+            data-testid={ `${ testId }-page-layout` }
+            headingColumnWidth={ 13 }
+            actionColumnWidth={ 3 }
+        >
+            <ApplicationTemplateGrid
+                onTemplateSelect={ (template: ExtensionTemplateListInterface) => {
+                    setSelectedTemplate(template);
+                    setShowWizard(true);
+                } }
+            />
+            <ApplicationTemplateProvider
+                template={ selectedTemplate }
+            >
+                <ApplicationTemplateMetadataProvider
+                    template={ selectedTemplate }
+                >
+                    <ApplicationCreationAdapter
+                        template={ selectedTemplate }
+                        showWizard={ showWizard }
+                        onClose={ () => setShowWizard(false) }
+                    />
+                </ApplicationTemplateMetadataProvider>
+            </ApplicationTemplateProvider>
+        </PageLayout>
+    );
+};
+
+/**
+ * A default export was added to support React.lazy.
+ * TODO: Change this to a named export once react starts supporting named exports for code splitting.
+ * @see {@link https://reactjs.org/docs/code-splitting.html#reactlazy}
+ */
```

Comment:
No need to keep this comment as we are purposely using default exports now.

Reply:
This is an existing code block. https://github.com/wso2/identity-apps/blob/05605b9367987c95ce06add63ce0e95494a90b2d/features/admin.applications.v1/pages/application-template.tsx#L528-L532